resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala

```diff
@@ -199,6 +200,37 @@ class ClientSuite extends SparkFunSuite with Matchers {
     appContext.getMaxAppAttempts should be (42)
   }
 
+  test("resource request for invalid resource") {
```

Comment:
Is this testing any Spark code at all? It seems to just be a long way to test YARN code that checks whether a resource type has been registered.

Reply:
Yes, this is very YARN-specific unlike the other testcases that verifies the resources in the app context.
Removed this testcase.
Do you think of any other way to test the scenario when a resource is specified in sparkConf which is not known for YARN?

Reply:
Okay, I agree with that. Then I think just removal is fine here.